### PR TITLE
Move session code into helper file

### PR DESCRIPTION
**Pull request by sergio-costas**
To reduce the code in main, the code used for checking if we are or not in a graphical session is moved into their own file. That's the only change done: move all the code for checking if the current session is a graphical one from `main` into its own source file.

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -11,6 +11,7 @@ snapd_desktop_integration = executable(
   'sdi-refresh-dialog.c',
   'sdi-refresh-monitor.c',
   'sdi-theme-monitor.c',
+  'sdi-user-session-manager.c',
```
Comment (robert-ancell) [inline src/meson.build:14]: Doesn't sound like a manager, as there is no class just one helper method. Not sure of a better name but perhaps something like `sdi-graphical-session-helper.c`.

**Review by robert-ancell — APPROVED** [review]
The name was unexpected but the code changes look good.

File: src/meson.build (hunk 1 above)
Comment (sergio-costas) [reply src/meson.build:14]: You are right. Changing it.

Comment (sergio-costas) [reply src/meson.build]: Done.

**Pull request merged**